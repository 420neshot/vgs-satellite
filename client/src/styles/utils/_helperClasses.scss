.flex-center {
  @include flexPosition(center, center);
}

.flex-left-center {
  @include flexPosition($align: center);
}

.flex-right-center {
  @include flexPosition(flex-end, center);
}

.flex-spc-btw-center {
  @include flexPosition(space-between, center);
}

.flex-spc-arnd-center {
  @include flexPosition(space-around, center);
}

.flex-spc-evn-center {
  @include flexPosition(space-evenly, center);
}

.flex-spc-btw-top {
  @include flexPosition(space-between, flex-start);
}

.flex-column-center {
  @include flexPosition(center, center, column);
}

.flex-column-left-center {
  @include flexPosition(center, flex-start, column);
}

.flex-column-right-center {
  @include flexPosition(center, flex-end, column);
}

.pointer-events-none {
  pointer-events: none;
}
