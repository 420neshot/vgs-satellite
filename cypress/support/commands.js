import 'cypress-file-upload';
import 'cypress-plugin-snapshots/commands';

// ***********************************************
// This example commands.js shows you how to
// create various custom commands and overwrite
// existing commands.
//
// For more comprehensive examples of custom
// commands please read more here:
// https://on.cypress.io/custom-commands
// ***********************************************
//
//
// -- This is a parent command --
// Cypress.Commands.add("login", (email, password) => { ... })
//
//
// -- This is a child command --
// Cypress.Commands.add("drag", { prevSubject: 'element'}, (subject, options) => { ... })
//
//
// -- This is a dual command --
// Cypress.Commands.add("dismiss", { prevSubject: 'optional'}, (subject, options) => { ... })
//
//
// -- This will overwrite an existing command --
// Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })

Cypress.Commands.add('cleanupRoutes', () => {
  cy.request('http://localhost:8089/route')
    .then((response) => {
      response.body.forEach((route) => {
        cy.request('DELETE', `http://localhost:8089/route/${route.id}`);
      });
    });
});

Cypress.Commands.add('cleanupFlows', () => {
  cy.request('http://localhost:8089/flows.json')
    .then((response) => {
      response.body.forEach((flow) => {
        cy.request('DELETE', `http://localhost:8089/flows/${flow.id}`);
      });
    });
});

// https://github.com/meinaart/cypress-plugin-snapshots/issues/10#issuecomment-610639709
Cypress.Commands.add('fixCypressSpec', filename => {
  const path = require('path');
  const relative = filename.substr(1); // removes leading "/"
  const projectRoot = Cypress.config('projectRoot');
  const absolute = path.join(projectRoot, relative);
  Cypress.spec = {
    absolute,
    name: path.basename(filename),
    relative,
  };
});
