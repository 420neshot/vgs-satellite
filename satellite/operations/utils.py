from contextlib import contextmanager
from dataclasses import dataclass, field
from time import monotonic, monotonic_ns


def monotonic_ms():
    return int(monotonic() * 1000)


@dataclass
class ExecutionTimeContext:
    start_ms: int = field(
        default_factory=lambda: monotonic_ms(),
        init=False,
    )
    start_ns: int = field(
        default_factory=lambda: monotonic_ns(),
        init=False,
    )
    elapsed_ms: int = field(default=None, init=False)
    elapsed_ns: int = field(default=None, init=False)


@contextmanager
def measure_execution_time() -> ExecutionTimeContext:
    ctx = ExecutionTimeContext()
    try:
        yield ctx
    finally:
        ctx.elapsed_ms = monotonic_ms() - ctx.start_ms
        ctx.elapsed_ns = monotonic_ns() - ctx.start_ns
