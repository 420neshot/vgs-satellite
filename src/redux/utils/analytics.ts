import config from 'src/config/config';

export const initAnalytics = () => {
  if (config.heapTrackingId) {
    // eslint-disable-next-line
    window.heap=window.heap||[],heap.load=function(e,t){window.heap.appid=e,window.heap.config=t=t||{};var r=document.createElement("script");r.type="text/javascript",r.async=!0,r.src="https://cdn.heapanalytics.com/js/heap-"+e+".js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(r,a);for(var n=function(e){return function(){heap && heap.push && heap.push([e].concat(Array.prototype.slice.call(arguments,0)))}},p=["addEventProperties","addUserProperties","clearEventProperties","identify","resetIdentity","removeEventProperty","setEventProperties","track","unsetEventProperty"],o=0;o<p.length;o++)heap[p[o]]=n(p[o])};
    window.heap.load(config.heapTrackingId);
  }
}

export const pushEvent = (eventName: string, eventProps?: { [key: string]: string }) => {
  window.heap?.track(eventName, eventProps || null);
};
