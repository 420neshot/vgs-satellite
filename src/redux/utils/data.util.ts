import _ from 'lodash';

export const XMLHeaders = [
  'application/xml',
  'text/xml',
  'application/xhtml+xml',
  'application/atom+xml',
];

export const JSONHeaders = [
  'application/json',
  'application/vnd.api+json',
];

export const FormDataHeaders = [
  'application/x-www-form-urlencoded',
];

export function isXML(contentType: string) {
  return _.some(XMLHeaders, header => contentType.startsWith(header));
}

export function isJSON(contentType: string) {
  return _.some(JSONHeaders, header => contentType.startsWith(header));
}

export function isFormData(contentType: string) {
  return _.some(FormDataHeaders, header => contentType.startsWith(header));
}

export function getTransformerType(contentType: string) {
  let transformer = 'UNKNOWN';

  if (isJSON(contentType)) {
    transformer = 'JSON_PATH';
  } else if (isXML(contentType)) {
    transformer = 'XPATH';
  } else if (isFormData(contentType)) {
    transformer = 'FORM_FIELD';
  }

  return transformer;
}
