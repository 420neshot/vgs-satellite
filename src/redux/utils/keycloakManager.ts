import Keycloak from 'keycloak-js';
import { pickBy } from 'lodash';
import { IKeycloakInstance } from 'src/redux/modules/auth';

const keycloakInstances: { [client: string]: IKeycloakInstance } = {};

const keycloakLogin = (kc: Keycloak.KeycloakInstance, loginOptions?: any) => {
  const url = window.location.href;
  const { origin } = window.location;
  const path = url.split(origin)[1];
  const exclusions = ['logout', '?idp=', 'session_state='];
  if (!exclusions.find((exclusion) => path.includes(exclusion)) && path !== '/') {
    localStorage.setItem('pathToRedirect', path);
  }
  const options = pickBy(
    {
      idpHint: new URL(window.location.href).searchParams.get('idp'),
      ...loginOptions,
    },
    (val: any) => Boolean(val),
  );

  return kc.login(options);
};

const init = (
  kcConfig: any,
  initParams: any,
  loginOptions?: any,
): Promise<IKeycloakInstance> => new Promise((res: any) => {
  keycloakInstances[kcConfig.clientId] = Keycloak(kcConfig) as IKeycloakInstance;

  keycloakInstances[kcConfig.clientId]
    .init(initParams)
    .then(async (isAuthenticated: boolean) => {
      if (isAuthenticated) {
        res(keycloakInstances[kcConfig.clientId]);
      } else {
        if (loginOptions) {
          keycloakLogin(keycloakInstances[kcConfig.clientId], loginOptions);
        }
      }
    })
    .catch(() => {
      localStorage.removeItem('authState');
      if (loginOptions) {
        keycloakLogin(keycloakInstances[kcConfig.clientId], loginOptions);
      }
    });
});

export default init;
