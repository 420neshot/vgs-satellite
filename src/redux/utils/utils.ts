import moment from 'moment';
import classnames from 'classnames';
import parseUrl from 'url-parse';
import { volatileRulesRegex } from 'src/data/regex';
import { ILog, ILogFilters } from 'src/redux/interfaces/logs';
import { IRoute, IEntry } from 'src/redux/interfaces/routes';
import randomId from './random-id';

export function formatDate(date) {
  if (moment().isSame(date, 'day')) {
    return moment.utc(date).local().format('[Today] HH:mm:ss');
  } else {
    return moment.utc(date).local().format('YYYY-MM-DD HH:mm:ss');
  }
}

export function dateFromNow(date) {
  return moment(date).fromNow();
}

export function nowBeforeDate(date) {
  return moment().isBefore(date);
}

export function isVolatile(name) {
  return name.match(volatileRulesRegex);
}

export function mapMatchingFilters(
  routes: IRoute[],
  logFilters: ILogFilters,
) {
  if (logFilters && logFilters.filters && !!logFilters.filters.length) {
    const matchingFilters = []
      .concat(
        ...routes.map(route =>
          logFilters.filters.map((f) => {
            const matchedFilter = route.entries.find(
              entry => entry.id === f.filter_id,
            );
            return matchedFilter && matchedFilter.id;
          }),
        ),
      )
      .filter(el => el != null);
    return matchingFilters;
  } else {
    return [];
  }
}

export function badgeColor(statusCode) {
  return classnames({
    danger: statusCode >= 500,
    success: statusCode >= 200 && statusCode <= 299,
    secondary: statusCode >= 300 && statusCode <= 399,
    warning: statusCode >= 400 && statusCode <= 499,
  });
}

export function flatExpressions(filterRule) {
  const allExpressions = (rule) => {
    const arr = [];
    if (!!rule.expression && rule.expression.field) {
      arr.push(rule.expression.field);
    }
    if (rule.rules && !!rule.rules.length) {
      rule.rules.forEach((innerRule) => {
        arr.push(allExpressions(innerRule));
      });
    }
    return arr;
  };

  return uniq(flattenDeep(allExpressions(filterRule)))
    .toString()
    .replace(/,/g, ', ');
}

export function matchedInActivePhase(entries: IEntry[], activePhase: string) {
  return entries.find(entry => entry.phase.toLowerCase() === activePhase);
}

export function isEqual(stringOne: string, stringTwo: string) {
  return stringOne.toLowerCase() === stringTwo.toLowerCase();
}

export function isInCollection(id: string, arr?: any[]) {
  if (arr && arr.find(obj => obj.id === id)) {
    return true;
  } else return false;
}

export function getQueryParameters(url: string) {
  return parseUrl(url).query;
}

export const generateRouteName = (hostname?: string) => {
  const routeName = randomId();
  // return hostname ? `${hostname}-${routeName}` : routeName;
  return routeName;
};

export function constructUriFromLog(log: ILog) {
  const isPathSufficient = !!log.path?.match(/^(https?:)\/\//);
  const isPathAppendable = !log.upstream?.match(/\/$/) && !!log.path?.match(/^\//);
  return isPathSufficient
    ? log.path
    : `${log.upstream ?? ''}${isPathAppendable ? log.path : ''}`;
}
