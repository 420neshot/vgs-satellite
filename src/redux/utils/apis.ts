import axios, { AxiosError, AxiosResponse } from 'axios';
import Qs from 'qs';
import trimFields from 'src/redux/utils/trim-attrs';
import { notify, FETCH_FAILED_ERROR_TEXT } from 'src/redux/utils/notifications';
import _ from 'lodash';
import { isJSON } from 'src/redux/utils/data.util';
import history from 'src/redux/utils/history';
import { refreshIfExpired } from 'src/redux/modules/auth';

interface IHeadersOverride { [key: string]: string | boolean | number; }
interface IFetchHandlers { onSuccess?: (res: AxiosResponse) => any; onError?: (err: AxiosError) => any; }
interface IOptions { headers?: IHeadersOverride; handlers?: IFetchHandlers; }
interface IModel {
  id: string;
  attributes: Record<string, string>;
  links?: Record<string, string>;
  type?: string;
  included?: IModel[];
  meta?: Record<string, string>;
  relationships?: Record<string, { links: {related: string, self: string} }>;
}
interface IModelJSON extends Omit<IModel, 'id' | 'attributes' | 'included' >{
  data?: {id: IModel['id'], [key: string]: string };
  included?: IModelJSON[];
}

const VGSApi = axios.create({
  paramsSerializer(params) {
    return Qs.stringify(params, { arrayFormat: 'brackets', encode: false });
  },
});

export const fetchApi = async (url: string, options: any) => {
  let params = { ...options, url };
  if (options && !options.noTrim) params = trimFields({ ..._.cloneDeep(options), url });

  if (params.body) {
    params.body = JSON.stringify(params.body);
  }

  if (params.headers && params.headers.Authorization === 'Bearer [TOKEN]') {
    params.headers.Authorization = `Bearer ${await refreshIfExpired(params.headers['[CLIENT]'])}`;
    delete params.headers['[CLIENT]'];
  }

  return VGSApi.request(params);
};

export function modelToJson(model: IModel, noTrim = false): IModelJSON {
  if (model) {
    if (!noTrim) model.attributes = trimFields(model.attributes);
    return {
      data: { ...model.attributes, id: model.id },
      links: model.links,
      type: model.type,
      included: model.included?.map((m: IModel) => modelToJson(m, noTrim)) || [],
      meta: model.meta || {},
      relationships: model.relationships || {},
    };
  }
  return {};
}

export function handleRequestSuccess(response: AxiosResponse, { noResponseTrim: noTrim = false } = {}) {
  if (Array.isArray(response.data.data)) {
    if (response.data.data.length > 0 && response.data.data[0].type === 'vaults') {
      return {
        data: response.data.data.map((m: IModel) => modelToJson(m, noTrim)),
        meta: response.data.meta,
        links: response.data.links,
      };
    }
    if (response.data.data.length > 0 && response.data.data[0].attributes) {
      return {
        data: response.data.data.map((m: IModel) => ({ ...m.attributes, id: m.id })),
        meta: response.data.meta,
        links: response.data.links,
      };
    }
    return {
      data: response.data.data,
      meta: response.data.meta,
      links: response.data.links,
    };
  }
  return modelToJson(
    {
      ...response.data.data,
      included: response.data.included,
      links: response.data.data
        ? response.data.data.links
        : response.data.links,
    },
    noTrim,
  );
}

// TODO check whether handleRequestError used with non-axios request functions
export function handleRequestError(error: AxiosError & {status?: number, message?: string}) {
  if (error.response) {
    let errorMessage;

    const errorResponse = _.get(error, 'response.data.errors[0]') || _.get(error, 'message');

    if (errorResponse && errorResponse.code || (errorResponse.status && errorResponse.title && errorResponse.detail)) {
      // handling for JSONApi style
      if (isJSON(errorResponse.detail)) {
        errorResponse.detail = JSON.parse(errorResponse.detail)[0].errorMessage;
      }
      errorMessage = {
        title: `${errorResponse.status || errorResponse.code} ${errorResponse.title}`,
        message: errorResponse.detail,
      };
    } else if (_.get(error, 'response.data.message')) {
      // handling for 'JSONApi' style
      errorMessage = error.response.data.message;
    } else {
      // fallback
      errorMessage = FETCH_FAILED_ERROR_TEXT;
    }

    notify.error(errorMessage);
  } else {
    notify.error(typeof error.request === 'string'
      ? error.request
      : error.message);
  }

  if (error.status === 401 || error.response?.status === 401) {
    setTimeout(
      () => history.push('/logout'),
      1000,
    );
  }

  // eslint-disable-next-line @typescript-eslint/no-throw-literal
  throw error.message || JSON.stringify(error);
}

export const fetchJSONApi = (
  url: string,
  options: any,
  handlers?: IFetchHandlers,
) => {
  const { noResponseTrim = false } = options || {};
  const handleRequestSuccessOpts = (response: AxiosResponse) => handleRequestSuccess(response, { noResponseTrim });
  return fetchApi(url, options)
    .then(handlers?.onSuccess || handleRequestSuccessOpts)
    .catch(handlers?.onError || handleRequestError);
};

export function makeHeaders(override?: IHeadersOverride) {
  const headers: any = {
    'Content-Type': 'application/vnd.api+json',
    Accept: 'application/vnd.api+json',
    Authorization: 'Bearer [TOKEN]',
    ...override,
  };
  return headers;
}

export const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));

export const getData = (url: string, options: IOptions = {}) => fetchJSONApi(url, { method: 'GET', headers: makeHeaders(options.headers) }, options.handlers);
export const postData = (url: string, data: object, options: IOptions = {}) => fetchJSONApi(url, { data: { data }, method: 'POST', headers: makeHeaders(options.headers) }, options.handlers);
export const putData = (url: string, data: object, options: IOptions = {}) => fetchJSONApi(url, { data: { data }, method: 'PUT', headers: makeHeaders(options.headers) }, options.handlers);
export const patchData = (url: string, data: object, options: IOptions = {}) => fetchJSONApi(url, { data: { data }, method: 'PATCH', headers: makeHeaders(options.headers) }, options.handlers);
export const deleteData = (url: string, data: object = {}, options: IOptions = {}) => fetchJSONApi(url, { data, method: 'DELETE', headers: makeHeaders(options.headers) }, options.handlers);
