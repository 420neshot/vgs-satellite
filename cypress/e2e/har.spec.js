import { cutLine } from '../support/utils';
const fixtures = require('../fixtures/diffs/har.js');

describe('Localhoste upload single har flow', function() {
  beforeEach(() => {
    cy.intercept('GET', 'flows.json').as('getFlows');
    cy.intercept('GET', 'route').as('getRoutes');
    cy.cleanupFlows();
    cy.cleanupRoutes();
    cy.fixCypressSpec(__filename);
  });

  it('Visits Localhoste and gets 2 yamls from uploaded har file', function() {
    cy.visit('/');
    cy.get('.menu-item').contains('Logs').click();
    cy.wait(['@getRoutes', '@getFlows']);
    cy.get('[data-role="import-from-har"]').attachFile('upload-post.har');

    cy.contains('/post');

    cy.get('[data-role="logs-row"]').click();
    cy.contains('http://httpbin.org/post');
    cy.get('[data-role="log-details-modal-content"]').toMatchSnapshot();

    cy.get('[data-role="select-response-phase"]').click();
    cy.get('[data-role="select-request-phase"]').click();

    cy.get('[data-role="tab-headers"]').click();
    cy.get('[data-role="log-details-modal-content"]').toMatchSnapshot();

    cy.get('[data-role="select-response-phase"]').click();
    cy.get('[data-role="log-details-modal-content"]').toMatchSnapshot();
    cy.get('[data-role="select-request-phase"]').click();

    cy.get('[data-role="tab-body"]').click();
    cy.get('[data-role="log-payload-body-diff"]').should(($div) => {
      expect($div.get(0).innerText).to.eq(fixtures.bodyRequest);
    })

    cy.get('[data-role="select-response-phase"]').click();
    cy.get('[data-role="log-payload-body-diff"]').should(($div) => {
      expect($div.get(0).innerText).to.eq(fixtures.bodyResponse);
    })
    cy.get('[data-role="select-request-phase"]').click();
    cy.contains('{"foo": "bar"}');

    cy.get('.ant-btn-primary[data-role="secure-payload-btn"]').click();
    cy.get('[data-role="quick-integration-modal"]').toMatchSnapshot();
    cy.contains('foo: bar').click({ force: true });

    cy.get('[data-role="select-secure-payload"]').click();
    cy.get('.ant-modal-content .ant-modal-title').contains('Save route');
    cy.get('.ant-modal-content .nav-tabs .active').contains('Inbound');
    cy.get('.ant-modal-content .active [data-role="inbound-code-container"]');

    cy.fixture('upload-inbound.yaml').then(yaml => {
      cy.get('[data-role="inbound-code-container"] pre code').should($div => {
        expect(cutLine($div.get(0).innerText)).to.eq(cutLine(yaml));
      });
    });
    cy.contains('Export Inbound YAML').click();

    cy.get('.nav-link')
      .contains('Outbound')
      .click();

    cy.get('.ant-modal-content .nav-tabs .active').contains('Outbound');
    cy.get('.ant-modal-content .active [data-role="outbound-code-container"]');

    cy.fixture('upload-outbound.yaml').then(yaml => {
      cy.get('[data-role="outbound-code-container"] pre code').should($div => {
        expect(cutLine($div.get(0).innerText)).to.eq(cutLine(yaml));
      });
    });
    cy.contains('Export Outbound YAML').click().type('{esc}');
  });
});
