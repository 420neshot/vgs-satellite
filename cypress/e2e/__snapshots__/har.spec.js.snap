exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #0`] = `
<div data-role="log-details-modal-content">
  <nav class="nav-tabs nav-tabs-sm">
    <a href="#" class="active capitalize" data-role="tab-general">general</a
    ><a href="#" class="capitalize" data-role="tab-headers">headers</a
    ><a href="#" class="capitalize" data-role="tab-body">body</a>
  </nav>
  <div class="text-center mt-3 mb-2 position-relative">
    <button
      type="button"
      class="ant-btn d-flex position-absolute right-0 ant-btn-primary ant-btn-sm"
    >
      <span>Secure this payload</span>
    </button>
    <div role="group" class="btn-group">
      <button
        type="button"
        data-role="select-request-phase"
        class="active capitalize btn btn-primary-light btn-sm"
      >
        request</button
      ><button
        type="button"
        data-role="select-response-phase"
        class="capitalize btn btn-outline btn-sm"
      >
        response
      </button>
    </div>
    <hr class="my-3 p-0" />
  </div>
  <div class="mb-3 pt-0">
    <table class="general-table pb-3 table">
      <tbody>
        <tr>
          <td class="smallbold">Protocol</td>
          <td><span>HTTP/1.1</span></td>
        </tr>
        <tr>
          <td class="smallbold">URL</td>
          <td>http://httpbin.org/post</td>
        </tr>
        <tr>
          <td class="smallbold">Request Method</td>
          <td><span class="d-flex text-left align-items-center">POST</span></td>
        </tr>
        <tr>
          <td class="smallbold">Status Code</td>
          <td><span class="badge badge-success">200</span></td>
        </tr>
      </tbody>
    </table>
    <hr class="my-3" />
    <div class="matching-details mx-3">
      <h3 class="my-2">Match Information</h3>
      <p class="text-text-light">Nothing matched</p>
    </div>
  </div>
</div>
`;

exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #1`] = `
<div data-role="log-details-modal-content">
  <nav class="nav-tabs nav-tabs-sm">
    <a href="#" class="capitalize" data-role="tab-general">general</a
    ><a href="#" class="active capitalize" data-role="tab-headers">headers</a
    ><a href="#" class="capitalize" data-role="tab-body">body</a>
  </nav>
  <div class="text-center mt-3 mb-2 position-relative">
    <button
      type="button"
      class="ant-btn d-flex position-absolute right-0 ant-btn-primary ant-btn-sm"
    >
      <span>Secure this payload</span>
    </button>
    <div role="group" class="btn-group">
      <button
        type="button"
        data-role="select-request-phase"
        class="active capitalize btn btn-primary-light btn-sm"
      >
        request</button
      ><button
        type="button"
        data-role="select-response-phase"
        class="capitalize btn btn-outline btn-sm"
      >
        response
      </button>
    </div>
    <hr class="my-3 p-0" />
  </div>
  <div class="mt-3 pt-0">
    <section class="request mt-2">
      <table class="header-table table table-striped">
        <tbody>
          <tr class="">
            <td class="header-name">Host</td>
            <td class="header-original">httpbin.org</td>
          </tr>
          <tr class="">
            <td class="header-name">User-Agent</td>
            <td class="header-original">curl/7.64.1</td>
          </tr>
          <tr class="">
            <td class="header-name">Accept</td>
            <td class="header-original">*/*</td>
          </tr>
          <tr class="">
            <td class="header-name">Proxy-Connection</td>
            <td class="header-original">Keep-Alive</td>
          </tr>
          <tr class="">
            <td class="header-name">Content-Type</td>
            <td class="header-original">application/json</td>
          </tr>
          <tr class="">
            <td class="header-name">Content-Length</td>
            <td class="header-original">14</td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>
</div>
`;

exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #2`] = `
<div data-role="log-details-modal-content">
  <nav class="nav-tabs nav-tabs-sm">
    <a href="#" class="capitalize" data-role="tab-general">general</a
    ><a href="#" class="active capitalize" data-role="tab-headers">headers</a
    ><a href="#" class="capitalize" data-role="tab-body">body</a>
  </nav>
  <div class="text-center mt-3 mb-2 position-relative">
    <button
      type="button"
      class="ant-btn d-flex position-absolute right-0 ant-btn-primary ant-btn-sm"
    >
      <span>Secure this payload</span>
    </button>
    <div role="group" class="btn-group">
      <button
        type="button"
        data-role="select-request-phase"
        class="capitalize btn btn-outline btn-sm"
      >
        request</button
      ><button
        type="button"
        data-role="select-response-phase"
        class="active capitalize btn btn-primary-light btn-sm"
      >
        response
      </button>
    </div>
    <hr class="my-3 p-0" />
  </div>
  <div class="mt-3 pt-0">
    <section class="request mt-2">
      <table class="header-table table table-striped">
        <tbody>
          <tr class="">
            <td class="header-name">Date</td>
            <td class="header-original">Wed, 26 Feb 2020 15:22:46 GMT</td>
          </tr>
          <tr class="">
            <td class="header-name">Content-Type</td>
            <td class="header-original">application/json</td>
          </tr>
          <tr class="">
            <td class="header-name">Content-Length</td>
            <td class="header-original">464</td>
          </tr>
          <tr class="">
            <td class="header-name">Connection</td>
            <td class="header-original">keep-alive</td>
          </tr>
          <tr class="">
            <td class="header-name">Server</td>
            <td class="header-original">gunicorn/19.9.0</td>
          </tr>
          <tr class="">
            <td class="header-name">Access-Control-Allow-Origin</td>
            <td class="header-original">*</td>
          </tr>
          <tr class="">
            <td class="header-name">Access-Control-Allow-Credentials</td>
            <td class="header-original">true</td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>
</div>
`;

exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #3`] = `
<div
  data-role="quick-integration-modal"
  class="modal-dialog modal-lg"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Secure a Payload</h5>
      <span id="" class="cursor-pointer"><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <div>
        <h4 class="mb-3">
          Select the fields that you want to Redact<sup
            ><i
              aria-label="icon: question-circle"
              tabindex="-1"
              class="anticon anticon-question-circle d-inline-flex mr-2 text-text text-sm"
              style="font-size: 14px;"
              ><svg
                viewBox="64 64 896 896"
                focusable="false"
                class=""
                data-icon="question-circle"
                width="1em"
                height="1em"
                fill="currentColor"
                aria-hidden="true"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 0 0-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
                ></path></svg></i></sup
          >or Reveal<sup
            ><i
              aria-label="icon: question-circle"
              tabindex="-1"
              class="anticon anticon-question-circle d-inline-flex mr-2 text-text text-sm"
              style="font-size: 14px;"
              ><svg
                viewBox="64 64 896 896"
                focusable="false"
                class=""
                data-icon="question-circle"
                width="1em"
                height="1em"
                fill="currentColor"
                aria-hidden="true"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 0 0-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
                ></path></svg></i
          ></sup>
        </h4>
        <div>
          <div>
            <ul class="css-5q6xdk">
              <li class="css-79elbk">
                <div>
                  <input name="isGoing" type="checkbox" readonly="" />
                  <div
                    title="$.foo"
                    data-role="checkbox-text"
                    class="w-95 text-word-break"
                    style="display: inline-block; vertical-align: top; color: inherit; padding-left: 10px; cursor: pointer;"
                  >
                    foo: bar
                  </div>
                </div>
                <div><ul class="css-1s1o8u8" style="display: block;"></ul></div>
              </li>
            </ul>
            <div class="mt-5 text-center">
              <button
                type="button"
                disabled=""
                data-role="select-secure-payload"
                class="button-next btn btn-primary disabled"
              >
                Secure this payload
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
